creator-node/scripts/user_metadata_to_cnode_missing_files.js

```diff
@@ -0,0 +1,334 @@
+    for (let file of cnodeUser.files) {
+      // skip dir types for file fetching and writing, but insert these into db
+      if (file.type === 'dir') continue
+
+      console.log(`about to fetch file from user metadata: ${file.multihash}`)
+      const responseStream = await _getFileFromUserMetadata(file.multihash)
+
+      if (responseStream) {
+        await _writeStreamToFileSystem(responseStream, path.join('files_um', file.multihash))
+      } else {
+        console.log(`could not download file for object`, file)
+        writesSuccessful = false
+      }
+    }
+
+    // this return is in a for loop, but we're assuming data.cnodeUsers is an object with a single property
+    return { writesSuccessful, files: cnodeUser.files || [] }
+  }
+}
+
+/**
+ * Create the dir on the kube container volume and transfer the files to the appropriate file storage directory
+ */
+async function copyFilesToPrimary (wallet, files) {
+  for (let file of files) {
+    try {
+      const mkdirCmd = _mkdirPathCommand(file.storagePath)
+      if (!COMMANDS_RUN.has(mkdirCmd)) {
+        await _runCommand(mkdirCmd)
+        COMMANDS_RUN.add(mkdirCmd)
+      }
+      const sendFileCmd = _sendFileCommand(file.multihash, file.storagePath)
+      if (!COMMANDS_RUN.has(sendFileCmd)) {
+        await _runCommand(sendFileCmd)
+        COMMANDS_RUN.add(sendFileCmd)
+      }
+    } catch (e) {
+      throw new Error(`Error copying files to primary for wallet: ${wallet}, ${e.message}`)
+    }
+  }
+}
+
+/**
+ * Given a user's id, find and return the primary creator node and wallet address
+ */
+async function findUsersPrimary (userId) {
+  const resp = await axios({
+    method: 'get',
+    baseURL: METADATA_NODE_ENDPOINT,
+    url: '/users',
+    params: { id: userId },
+    responseType: 'json',
+    timeout: 300000 /* 5m = 300000ms */
+  })
+
+  if (resp.status !== 200) {
+    console.error(`Failed to retrieve findUsersPrimary`, userId)
+    throw new Error(resp.data['error'])
+  }
+
+  // resp.data is {data: [{user object}], signature: '', signer: ''}
+  if (!resp.data || !resp.data.data || !resp.data.data[0] || !resp.data.data[0].creator_node_endpoint) {
+    throw new Error(`Malformed response for findUsersPrimary ${userId}`)
+  }
+
+  const primary = resp.data.data[0].creator_node_endpoint.split(',')[0]
+  const wallet = resp.data.data[0].wallet
+  if (primary && wallet) {
+    console.log(`primary endpoint for wallet ${wallet}: ${primary}`)
+    return { primary, wallet }
+  }
+
+  return null
+}
+
+async function writeFilesToDB (wallet, files) {
+  // find all existing files and dedupe
+  // create a tx
+  // in a try/catch, call dbManager function to write files
+  let cnodeUserUM
+  let cnodeUserCN
+  let filesToAdd = []
+  try {
+    cnodeUserUM = await UM_MODELS.CNodeUser.findOne({ where: { walletPublicKey: wallet } })
+    cnodeUserCN = await CN_MODELS.CNodeUser.findOne({ where: { walletPublicKey: wallet } })
```

Comment:
@SidSethi ironically the cnodeUserUUID between UM and CN is different here. going forward with your diff based sync change where cnodeUserUUID won't be copied over is actually correct